lib/srv/alpnproxy/local_proxy.go

```diff
@@ -129,6 +146,14 @@ func (l *LocalProxy) Start(ctx context.Context) error {
 			log.WithError(err).Errorf("Failed to accept client connection.")
 			return trace.Wrap(err)
 		}
+
+		if l.cfg.Middleware != nil {
+			if err := l.cfg.Middleware.OnNewConnection(ctx, l, conn); err != nil {
+				log.WithError(err).Errorf("Middleware failed to handle new connection.")
+				continue
+			}
```

Comment:
I'm currently implementing a mini version of what you have in `local_proxy_middleware.go`, a middleware that specifically handles expired certs only (no MFA support yet) and reaches out to Connect to trigger the login modal.

If `l.cfg.Middleware.OnNewConnection` returns an error, from the client's perspective the connection just hangs indefinitely. Should we close the connection on error instead?

Reply:
The reason I did continue was because the connection is passed into the middleware to decide what to do. Although I did forget to close the connection on error in the middleware...

Reply:
So I agree we should update this, either to close connection here or to handle the error by closing the connection in the middleware

Reply:
I think we should just close it here unless we find a reason to not close the connection if the middleware encounters an error.

---

BTW, for now the middleware I'm adding will be super simple as it'll only check for the expired cert and nothing more. In the future though, once we get around to finally adding per-session MFA support to Connect, I plan to refactor the existing middleware a bit so that it can be shared between Connect and tsh.

From what I gathered, the logic will be mostly the same and the two will only differ (as usual) in how they handle UI interactions (CLI vs GUI).